mpy-cross/main.c

```diff
@@ -312,6 +313,9 @@ MP_NOINLINE int main_(int argc, char **argv) {
                 } else if (strcmp(arch, "xtensawin") == 0) {
                     mp_dynamic_compiler.native_arch = MP_NATIVE_ARCH_XTENSAWIN;
                     mp_dynamic_compiler.nlr_buf_num_regs = MICROPY_NLR_NUM_REGS_XTENSAWIN;
+                } else if (strcmp(arch, "rv32i") == 0) {
+                    mp_dynamic_compiler.native_arch = MP_NATIVE_ARCH_RV32I;
+                    mp_dynamic_compiler.nlr_buf_num_regs = MICROPY_NLR_NUM_REGS_RV32I;
```

Comment:
I would call the architecture `rv32imc` to indicate the extensions required.  This will allow room to grow in the future if we also need to support strict `rv32i` without extensions.

Reply:
Makes sense, although I wonder whether there are recently produced MCUs that don't have the M extension present.  Except soft-cores, that is, but that doesn't really count, does it?

Anyway, I'll add the platform renaming to the new batch of commits.

Reply:
You may be right that everything has M extension support, and probably also C.  But at this stage (early stage for RISC-V lifetime) it's hard to tell and so I'd go conservative and spell it out as `rv32imc`.

And supporting soft-cores is definitely in scope for MicroPython.